### ISSUE TITLE
No Instance AllCTRender/FromHttpApiData from using `prepareRoutes`

### ISSUE THREAD

**Issue by Cmdv** [aesiniath/unbeliever#92]
I get this error with API types are of a specific shape, when using with prepareRoutes 
```
     • No instance for (Servant.API.ContentTypes.AllCTRender
                          '[] Servant.API.ContentTypes.NoContent)
         arising from a use of 'prepareRoutes'
     • In the expression: prepareRoutes _API
       In a stmt of a 'do' block:
         application <- prepareRoutes _API $ routeHandlers runtime
       In the expression:
         do info "Program initiating..."
            port <- withDefaultOptionValue @Int 8080 "port"
            application <- prepareRoutes _API $ routeHandlers runtime
            info "Launching server"
            ....
    |
 28 |     application <- prepareRoutes _API $ routeHandlers runtime
    | 
                       ^^^^^^^^^^^^^^^^^^
```

These routes seem to work fine:
```Haskell
type API =
 ReqBody '[JSON] ATypeA :> Post '[JSON] ATypeB

```

but when one tries to complicate things slightly it seems to fall over eg this is what created above error:
```Haskell
type API =
 "test" :> Capture "id" Text :> Post '[] NoContent
```
Also seen it if the `Capture` type is a custom type like:
```Haskell
type MyID = MyID Text
-- or
newtype MyID = MyID { unMyID : Text }
...
Capture "id" MyID :> Post '[] Int
```

then the error is:
_looking at this more think this might be expected behaviour but it was back in 2017 so it's hard to tell, I've seen examples with custom `Capture` types_ [this issue](https://github.com/haskell-servant/servant/issues/708)

```
No instance for http-api-data-0.4.2:Web.Internal.HttpApiData.FromHttpApiData
                          Call.Backend.Auth0.Types.Authentication.UserInfo.UserId
```

**Comment by and-pete** [comment]
👋
This error:
```yaml
     • No instance for (Servant.API.ContentTypes.AllCTRender
                          '[] Servant.API.ContentTypes.NoContent)
         arising from a use of 'prepareRoutes'
```         
...comes from the empty content type lists in both:
```haskell
Post '[] NoContent
Post '[] Int
```
when you give the proxy for that API type to Servant's `serve` (or `serveWithContext`) functions.

You'll see it goes away in both cases if you add in one (or more) of the [content type symbols](https://docs.servant.dev/en/stable/tutorial/Server.html#the-truth-behind-json):
```haskell
--
Post '[JSON] NoContent
Post '[PlainText] NoContent
Post '[JSON] Int
```
In the `Post`/`NoContent` case though, you'd probably want to return a `204` instead of the `200`. Servant has endpoint verbs like `PostNoContent` (see [here](https://github.com/haskell-servant/servant/blob/c388c5e82c3df5d8ae63b523fea012c23e37410a/servant/src/Servant/API/Verbs.hs#L123-L124)) and equivalent for other verbs to help you out with that.
e.g.
```haskell
type API =
 "test" :> Capture "id" MyId :> PostNoContent
 ```
 In the case of the FromHttpApIData error, see how they write one in the Servant tutorial docs [here](https://docs.servant.dev/en/stable/tutorial/Server.html#the-fromhttpapidata-tohttpapidata-classes).
 Or depending on your needs, it's easily derived:
 ```haskell
 newtype MyId = MkMyId { getMyId :: Text }
   deriving stock Show
   deriving newtype FromHttpApiData
 ```

**Comment by Cmdv** [comment]
@and-pete 😍 Thanks a lot, that error was really throwing me. Didn't realise you couldn't have an empty content type. Really thought I'd seen it in the wild.

Interesting in regards to the `PostNoContent`. I did have a play with the `Verb` too, see if that made a difference.
I've bee learning a lot more in with Servant as prior it was mostly Yesod and that slightly abstracts these parts away by generating the whole API from a file.

I'll try this out then close it once I've implemented it. Thanks again @and-pete!! @istathar give this man a job 😅
